src/utils/ScriptTransformer.ts

```diff
@@ -0,0 +1,159 @@
+import * as babel from "@babel/standalone";
+import { transformSync, type ParserConfig } from "@swc/wasm-web";
+import * as acorn from "acorn";
+import { resolveScriptFilePath, validScriptExtensions, type ScriptFilePath } from "../Paths/ScriptFilePath";
+import type { Script } from "../Script/Script";
+
+// This is only for testing. It will be removed after we decide between Babel and SWC.
+declare global {
+  // eslint-disable-next-line no-var
+  var forceBabelTransform: boolean;
+}
+
+export type AcornASTProgram = acorn.Program;
+export type BabelASTProgram = object;
+export type AST = AcornASTProgram | BabelASTProgram;
+
+export enum FileType {
+  PLAINTEXT,
+  JSON,
+  JS,
+  JSX,
+  TS,
+  TSX,
+  NS1,
+}
+
+export interface FileTypeFeature {
+  isReact: boolean;
+  isTypeScript: boolean;
+}
+
+export class ModuleResolutionError extends Error {}
+
+const supportedFileTypes = [FileType.JSX, FileType.TS, FileType.TSX] as const;
+
+export function getFileType(filename: string): FileType {
+  const extension = filename.substring(filename.lastIndexOf(".") + 1);
+  switch (extension) {
+    case "txt":
+      return FileType.PLAINTEXT;
+    case "json":
+      return FileType.JSON;
+    case "js":
+      return FileType.JS;
+    case "jsx":
+      return FileType.JSX;
+    case "ts":
+      return FileType.TS;
+    case "tsx":
+      return FileType.TSX;
+    case "script":
+      return FileType.NS1;
+    default:
+      throw new Error(`Invalid extension: ${extension}. Filename: ${filename}.`);
+  }
+}
+
+export function getFileTypeFeature(fileType: FileType): FileTypeFeature {
+  const result: FileTypeFeature = {
+    isReact: false,
+    isTypeScript: false,
+  };
+  if (fileType === FileType.JSX || fileType === FileType.TSX) {
+    result.isReact = true;
+  }
+  if (fileType === FileType.TS || fileType === FileType.TSX) {
+    result.isTypeScript = true;
+  }
+  return result;
+}
+
+export function parseAST(code: string, fileType: FileType): AST {
+  const fileTypeFeature = getFileTypeFeature(fileType);
+  let ast: AST;
+  if (fileType === FileType.JS) {
+    ast = acorn.parse(code, { sourceType: "module", ecmaVersion: "latest" });
```

Comment:
`acorn` is much faster than `babel-parser`, especially when parsing many big JS files. `babel-parser` is only useful when we have to parse JSX and TypeScript.

Reply:
Speaking of documentation, it would be good if some of the core design decisions (such as the rationale you just gave) were captured in code comments, for posterity.

Reply:
It might still make sense to rewrite the imports in babel, if we are going through babel any way. Something like https://github.com/tomprince/bitburner-src/commit/wip-typescript~1 maybe.